src/MemoryOptions.php

```diff
@@ -76,17 +89,17 @@ public function getMemoryLimit()
      * Normalized a given value of memory limit into the number of bytes
      *
      * @param string|int $value
-     * @throws Exception\InvalidArgumentException
      * @return int
+     * @throws Exception\InvalidArgumentException
      */
     protected function normalizeMemoryLimit($value)
     {
         if (is_numeric($value)) {
             return (int) $value;
         }
 
-        if (! preg_match('/(\-?\d+)\s*(\w*)/', ini_get('memory_limit'), $matches)) {
-            throw new Exception\InvalidArgumentException("Invalid  memory limit '{$value}'");
+        if (! preg_match('/^(\-?\d+)\s*(\w*)$/', $value, $matches)) {
```

Comment:
I think I'm fine with this one. I would say lets remove the `$` for now.
I still wonder why a negative numeric value would be okay here but who am I to judge this ancient code.
Most probably only @weierophinney or @marc-mabe are aware of that and thus lets keep it for now. Could be removed in next major.
I most probably keep that in mind until then.

Reply:
negative numeric is for -1 = no limit

Reply:
Yup, but thats how it worked before as well.
If s1 wants to set unlimited memory, thats fine in the current major as thats how it works since like forever.
I am open to discuss this in an RFC for the next major version, would you agree on that?